starter_code/elevator.js

```diff
@@ -1,19 +1,64 @@
+const Person = require('./person.js');
 class Elevator {
-  constructor(){
-    this.floor      = 0;
-    this.MAXFLOOR   = 10;
-    this.requests   = [];
+  constructor() {
+    this.floor = 0;
+    this.MAXFLOOR = 10;
+    this.requests = [];
+    this.waitingList = [];
+    this.passengers = [];
+    this.direction = "up"
   }
 
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  start() {
+    this.intervalId = setInterval(() => { this.update(); }, 1000)
+  }
+  stop() {
+    console.log('Elevator stop');
+    clearInterval(this.intervalId);
+  }
+  update() {
+    this.log();
+    this._passengersEnter();
+    this._passengersLeave();
+    if (this.floor === this.requests[0]) { this.requests.shift() }
```

Comment:
sí quieres poner un if en una linea también puedes hacerlo sin llaves 
if (condición) this.requests.shift()